src/rules.c

```diff
@@ -118,7 +118,7 @@ SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, IndexSpec *spec, QueryError
     rule->prefixes = array_append(rule->prefixes, p);
   }
 
-  rule->spec = spec;
+  rule->spec = StrongRef_Get(ref);
```

Comment:
Here we should also hold the weak or strong ref, right?

Reply:
I'm not sure. I left a TODO in `rules.h` to verify why the rule needs its spec

Reply:
I believe its a weak ref, see my comment bellow.

Reply:
check out #3352, it might be a better fix